modules/internal/ChapelDomain.chpl

```diff
@@ -1566,6 +1566,59 @@ module ChapelDomain {
       return _newArray(x);
     }
 
+    pragma "no copy return"
+    @unstable("tryCreateArray() is subject to change in the future.")
+    proc tryCreateArray(type eltType, initExpr: ?t) throws
+      where isSubtype(t, _iteratorRecord) || isCoercible(t, eltType) {
```

Comment:
I might add one more test that exercises assigning a default value to the array, rather than just an iterator or an array. Ex:
```chapel
var arr = dom.tryCreateArray(int, 42);
writeln(arr);
```